bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -289,16 +289,19 @@ public void run()
           }
           final LogicalNode oln = subscriberGroups.put(type, ln);
           if (oln != null) {
-            oln.boot(eventloop);
+            oln.boot();
           }
-          AbstractLengthPrependerClient subscriber = new Subscriber(ln, request.getBufferSize());
-
-          subscriber.registered(key);
-          key.attach(subscriber);
-          key.interestOps(SelectionKey.OP_WRITE | SelectionKey.OP_READ);
-
-          ln.catchUp();
-          dl.addDataListener(ln);
+          final Subscriber subscriber = new Subscriber(ln, request.getBufferSize());
+          eventloop.submit(new Runnable()
```

Comment:
Wait isn't the subscribe request already being run in the same event loop thread already?